File: CarlaErik/UvA.SoftCon.Questionnaire/Parsing/ShortToUnicodeString.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UvA.SoftCon.Questionnaire.Parsing
{
    public class ShortToUnicodeString : QLBaseListener
    {
        private StringBuilder _result = new StringBuilder();

        public override void EnterInit(QLParser.InitContext context)
        {
            _result.Append('"');
        }

        public override void ExitInit(QLParser.InitContext context)
        {
            _result.Append('"');
        }

        public override void EnterValue(QLParser.ValueContext context)
        {
            int value = Int32.Parse(context.INT().GetText());
            _result.AppendFormat(@"\u{0:X04}", value);
        }

        public string GetResult()
        {
            return _result.ToString();
        }
    }
}

File: Nian_Karl/QL_Nian/src/example1/ulti/ShortToUnicodeString.java
package example1.ulti;

import example1.antlr.ArrayInitBaseListener;
import example1.antlr.ArrayInitParser;

public class ShortToUnicodeString extends ArrayInitBaseListener{
	@Override public void enterInit(ArrayInitParser.InitContext ctx) {
		System.out.print('"');
	}
	
	@Override public void exitInit(ArrayInitParser.InitContext ctx) {
		System.out.print('"');
	}
	
	@Override public void exitValue(ArrayInitParser.ValueContext ctx) {
		int value = Integer.valueOf(ctx.INT().getText());
		System.out.printf("\\u%04x", value);
	}
}
